stdlib/public/Python/NumpyConversion.swift

```diff
@@ -152,41 +152,79 @@ extension Double : NumpyScalarCompatible {
   }
 }
 
-extension Array : ConvertibleFromNumpyArray
-  where Element : NumpyScalarCompatible {
+/// A helper struct with all the data that ConvertibleFromNumpyArray
+/// implementations need to copy memory from a NumpyArray.
+public struct ContiguousNumpyArray<Scalar, ShapeSize> {
```

Comment:
I don't see a reason to use a generic parameter `ShapeSize`. I can see that you intended to unify `Tensor`'s index type with `Array`'s index type, but it is not necessary. TensorFlow is not involved here, so `Int` should always be preferred.

Even for some reason if it's required, it should be renamed to `Index` for naming convention, and be constrained to `BinaryInteger & PythonConvertible` for its semantics and the precise use case.